core/src/mapping.rs

```diff
@@ -84,11 +90,10 @@ impl<K: ToBytes, V: ToBytes + FromBytes + CLTyped + OverflowingAdd + Default> Ma
     ///
     /// If the operation fails due to overflow, the currently executing contract reverts.
     pub fn add(&mut self, key: &K, value: V) {
-        let current_value = self.get_or_default(key);
-        let new_value = current_value
-            .overflowing_add(value)
-            .unwrap_or_revert(&self.env_for_key(key));
-        self.set(key, new_value);
+        let env = self.env_for_key(key);
+        let current_value = Var::<V>::instance(Rc::new(env.clone()), self.index).get_or_default();
```

Comment:
The intention was to calculate the key once, because the `get()` and `set()` methods call the `env.current_key()` function. But that implementation is wrong. I fixed it in the next commit.